### PR TITLE
Redirect to gotest.tools/v3 if no version is specified

### PR DESCRIPTION
**Pull request by dnephin**
https://github.com/gotestyourself/gotest.tools/issues/188#issuecomment-587569538

### CODE REVIEW

File: handler.go [hunk 1]
```diff
@@ -123,6 +127,15 @@ func (h *handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
 }
 
+func withDefaultVersion(version string, path string) string {
+	if version == "" || versionRegexp.MatchString(path) {
+		return path
+	}
+	return version + "/" + path
+}
+
+var versionRegexp = regexp.MustCompile(`^v\d`)
```
Comment (thaJeztah) [inline handler.go:137]: nit: might be clearer to move this above the function above

There's probably a corner-case for `v1`, which should not be matched

File: handler_test.go [hunk 2]
```diff
@@ -97,39 +100,80 @@ func TestHandler(t *testing.T) {
...
-		}
-		if got := findMeta(data, "go-import"); got != test.goImport {
-			t.Errorf("%s: meta go-import = %q; want %q", test.name, got, test.goImport)
-		}
-		if got := findMeta(data, "go-source"); got != test.goSource {
-			t.Errorf("%s: meta go-source = %q; want %q", test.name, got, test.goSource)
-		}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
```
Comment (thaJeztah) [inline handler_test.go:138]:
```suggestion
		tc := tc
		t.Run(tc.name, func(t *testing.T) {
```

**Review by thaJeztah — CHANGES_REQUESTED** [review]
some minor nit/suggestions, but otherwise looks ok

File: handler.go (hunk 1 above)
Comment (dnephin) [reply handler.go:137]: Why shouldn't `v1` be matched? I think if  there is any `v\d` at the start of the path we don't want to prepend a different version.

File: handler_test.go (hunk 2 above)
Comment (dnephin) [reply handler_test.go:138]: Capturing the loop variable is only necessary if `t.Parallel()` is used, right?

File: handler.go (hunk 1 above)
Comment (thaJeztah) [reply handler.go:137]: Train of through here was that Golang only uses `/vXX` for imports > v2, so v1 packages would be named without v1 in their name

Comment (dnephin) [reply handler.go:137]: Right. So with this change it would be impossible to get redirected to the v1 pages on http://pkg.go.dev/. It assumes that anyone typing in the url without a version wants the latest version. You can still go to the pre-v3 pages once you get there.

I don't see a way around that. I'd rather not add more logic for redirecting `v1`, since that is unlikely to be useful.
